example/router/router_test.cpp

```diff
@@ -0,0 +1,265 @@
+//
+// Copyright (c) 2022 Alan de Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/url
+//
+
+// Test that header file is self-contained.
+#include <router.hpp>
+#include <src.hpp>
+
+#include "test_suite.hpp"
```

Comment:
This file should probably go in test/example/router/router.cpp

Reply:
Oh... we can do that!